tests/app/main/views/test_suppliers.py

```diff
@@ -810,6 +810,21 @@ def test_should_set_declaration(self, data_api_client):
 
 
 @mock.patch('app.main.views.suppliers.data_api_client')
+@mock.patch('app.main.views.suppliers.download_agreement_file')
+class TestDownloadSignedAgreementFile(LoggedInApplicationTest):
+    user_role = 'admin-ccs-sourcing'
+
+    def test_download_agreement_is_called_with_the_right_parameters(self, download_agreement_file, data_api_client):
+        data_api_client.get_supplier_framework_info.return_value = {
+            'frameworkInterest': {'agreementPath': '/path/to/file/in/s3/1234-signed-agreement-file.pdf'}
+        }
+        # Mock out a response from download_agreement_file() - we don't care what it is
+        download_agreement_file.side_effect = HTTPError(Response(404))
```

Comment:
Maybe I haven't quite understood what's going on but if we don't care what the response is from `download_agreement_file`, do we even need to set a side effect? Putting an HTTPError here makes me think that the error is important/relevant to the test and we are testing error conditions when we aren't...